3/scripts.py

```diff
@@ -146,79 +169,106 @@ def get_neighbourhood_2n(solution, matrix):
             neighbor[i] = node
             delta = calculate_delta(solution, matrix, node, i)
             neighbors.append((neighbor, delta))
+
+    return neighbors
+
+
+def get_neighbourhood_2e(solution, matrix):
+    neighbors = []
+    solution_length = len(solution)
+    matrix_shape = matrix.shape[0]
+
+    # intra-route
+    for i in range(solution_length-1):
+        for j in range(i + 1, solution_length):
+            neighbor = np.concatenate((solution[:i], solution[i:j+1][::-1], solution[j+1:]))
+
+            delta = calculate_delta_edge(solution, matrix, i, j)
+            neighbors.append((neighbor, delta))
  
+    all_nodes = set(range(matrix_shape))
+    available_nodes = all_nodes - set(solution)
+
+    # inter-route
+    for i in range(solution_length):
+        for node in available_nodes:
+            neighbor = solution.copy()
+            neighbor[i] = node
+            delta = calculate_delta(solution, matrix, node, i)
+            neighbors.append((neighbor, delta))
+
     return neighbors
 
 
 def steepest_2n(matrix, starting_sol):
     best_sol = starting_sol
     best_delta = 0
     neighbourhood = get_neighbourhood_2n(starting_sol, matrix)
-    
+
     while len(neighbourhood):
         deltas = np.array([delta for _, delta in neighbourhood])
         best_index = np.argmin(deltas)
-        probably_best_sol, best_delta = neighbourhood[best_index]
+        best_sol, best_delta = neighbourhood[best_index]

```

Comment:
jak nie sprawdzimy delty przed przypisaniem solucji to możemy wziąć na końcu rozwiązanie gorsze od aktualnego